server/src/main/java/org/eclipse/openvsx/util/SnapshotHandler.java

```diff
@@ -0,0 +1,129 @@
+/********************************************************************************
+ * Copyright (c) 2023 Ericsson and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.openvsx.util;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.regex.Pattern;
+
+public class SnapshotHandler extends FileHandler {
+
+  private static final Pattern pattern = Pattern.compile("\\{.*\\}[\\r|\\n]");
+  // the following can be configured by Logging.properties
+  private final int maxEvents = 1000000;
+  private final double timeout = 30.0;
+  protected final String filePath = "request-";
+  // Enable scope logs by default
+  private volatile boolean fIsEnabled = true;
+
+  private Deque<InnerEvent> fData = new ArrayDeque<>();
+  private Map<String, Map<String, List<InnerEvent>>> fStacks = new HashMap<>();
+
+  public SnapshotHandler() throws IOException, SecurityException {
+    super();
+  }
+
+  @Override
+  public boolean isLoggable(LogRecord record) {
+    boolean logIt =
+        (fIsEnabled
+            && super.isLoggable(record)
+            && (record.getLevel().intValue() <= Level.FINE.intValue())
+            && (pattern.matcher(record.getMessage()).matches()));
+    if (logIt) { // add feature switch here
+      addToSnapshot(record.getMessage());
```

Comment:
I would say it is not a good idea at all to call any "command" methods in such "query"-by-design method as `isLoggable()`. The method should be read-only by all the specifications.

Reply:
Found the solution!

  public boolean isLoggable(LogRecord record) {
return 
        (fIsEnabled
            && super.isLoggable(record)
            && (record.getLevel().intValue() <= Level.FINE.intValue())
            && (pattern.matcher(record.getMessage()).matches()));
}

then override publish to put addToSnapshot there

Reply:
Fixed?